src/components/TestToolsModal.tsx

```diff
@@ -38,6 +38,10 @@ function TestToolsModal() {
     const isAuthenticated = useIsAuthenticated();
     const route = getRouteBasedOnAuthStatus(isAuthenticated, activeRoute);
 
+    if (!isTestToolsModalOpen) {
```

Comment:
I understand this is good to add, but was this for code quality or to fix the bug?